dash/include/dash/allocator/internal/Types.h

```diff
@@ -0,0 +1,71 @@
+#ifndef DASH__ALLOCATOR__INTERNAL__TYPES_H__INCLUDED
+#define DASH__ALLOCATOR__INTERNAL__TYPES_H__INCLUDED
+#include <cstddef>
+#include <ostream>
+
+namespace dash {
+namespace allocator {
+
+
+struct memory_block {
+  // Default Constructor
+  memory_block() noexcept
+    : ptr(nullptr)
+    , length(0)
+  {
+  }
+
+  memory_block(void *ptr, size_t length) noexcept
+    : ptr(ptr)
+    , length(length)
+  {
+  }
+
+  // Move Constructor
+  memory_block(memory_block &&x) noexcept { *this = std::move(x); }
```

Comment:
This looks quite strange. If this works, better use `memory_block(memory_block &&) = default;`